core/src/main/java/org/infinispan/remoting/rpc/RpcManagerImpl.java

```diff
@@ -107,23 +138,164 @@ public String getPendingViewAsString() {
    }
 
    @Override
+   public <T> CompletionStage<T> invokeCommand(Address target, ReplicableCommand command,
+                                               ResponseCollector<T> collector, RpcOptions rpcOptions) {
+      checkTopologyId(command);
+      CacheRpcCommand cacheRpc = toCacheRpcCommand(command);
+
+      if (!statisticsEnabled) {
+         return t.invokeCommand(target, cacheRpc, collector, rpcOptions.deliverOrder(),
+                                rpcOptions.timeout(), rpcOptions.timeUnit());
+      }
+
+      long startTimeNanos = timeService.time();
+      CompletionStage<T> invocation;
+      try {
+         invocation = t.invokeCommand(target, cacheRpc, collector, rpcOptions.deliverOrder(),
```

Comment:
I have already decided to do that, but there are some leftovers from `invokeRemotelyAsync` lying around.